### ISSUE TITLE
Support LLVM5

### ISSUE THREAD

**Issue by akzhan** [crystal-lang/crystal#4423]
MacOS example:

```bash
brew install llvm --HEAD
make deps
```

A lot of **ambiguous unwrap** errors.

```
Using /usr/local/bin/llvm-config [version=5.0.0svn]
c++ -c  -o src/llvm/ext/llvm_ext.o src/llvm/ext/llvm_ext.cc `/usr/local/bin/llvm-config --cxxflags`
src/llvm/ext/llvm_ext.cc:31:1: error: call to 'unwrap' is ambiguous
DEFINE_ISA_CONVERSION_FUNCTIONS(Metadata, LLVMMetadataRef)
^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/local/Cellar/llvm/HEAD-c2b55b1_1/include/llvm/Support/CBindingWrapping.h:34:20: note: expanded from
      macro 'DEFINE_ISA_CONVERSION_FUNCTIONS'
    return cast<T>(unwrap(P));                          \
                   ^~~~~~
/usr/local/Cellar/llvm/HEAD-c2b55b1_1/include/llvm/IR/Metadata.h:138:1: note: candidate function
DEFINE_ISA_CONVERSION_FUNCTIONS(Metadata, LLVMMetadataRef)
^
/usr/local/Cellar/llvm/HEAD-c2b55b1_1/include/llvm/Support/CBindingWrapping.h:30:3: note: expanded from
      macro 'DEFINE_ISA_CONVERSION_FUNCTIONS'
  DEFINE_SIMPLE_CONVERSION_FUNCTIONS(ty, ref)           \
  ^

...
```

**Comment by ysbaddaden** [comment]
You may either:

- fix support for LLVM 5
- use released LLVM 4

**Comment by RX14** [comment]
I don't think closing is an appropriate response, LLVM suggests that languages track LLVM HEAD and proactively follow their API changes. Regardless, this issue could be left open as a "support LLVM 5" issue.

**Comment by ysbaddaden** [comment]
Proactively following HEAD would require to regularly fix the API they broke. I won't look at broken changes until the LLVM5 beta is about to be released, and release notes updated. Right now it doesn't list any changes in DIBuilder...

There is no need to keep a "LLVM head is broken" issue opened. It would never get closed :(

**Comment by ysbaddaden** [comment]
Reopening, the LLVM 5 release candidate has been released, there shouldn't be more breaking changes until the official release.

**Comment by asterite** [comment]
Fixed in #4821
